### ISSUE TITLE
Weird problems when using in node app which uses AWS sdk

### ISSUE THREAD

**Issue by antonosmond** [node-inspector/node-inspector#703]
Not sure if this is a feature of node-inspector or a bug, but it appears to interfere with outgoing http requests, noticed when using the aws-sdk.
If I create the file app.js:
```
var AWS = require('aws-sdk');

var ec2 = new AWS.EC2({ region: 'us-east-1' });

ec2.describeVpcs({}, function(err, data) {
  console.log(err);
  console.log(data);
});
```
I make sure my AWS credentials are correctly set for my environment.
When I run `node --debug app.js`, there is no error and `data` contains an AWS response object with an array of my VPCs in us-east-1, like below:
```
{ 
  Vpcs:[ 
    { 
      VpcId: 'vpc-11111111',
      State: 'available',
      CidrBlock: '0.0.0.0/16',
      DhcpOptionsId: 'dopt-aaaaaaaa',
      Tags: [Object],
      InstanceTenancy: 'default',
      IsDefault: false
    },
    { 
      VpcId: 'vpc-22222222',
      State: 'available',
      CidrBlock: '1.1.1.1/16',
      DhcpOptionsId: 'dopt-bbbbbbbb',
      Tags: [Object],
      InstanceTenancy: 'default',
      IsDefault: false
    }
  ]
}
```
But if I run `node-debug app.js` and allow the app to run using node-inspector, I don't get an error, but I don't get my array of VPCs. `data` looks like an AWS response object, but the VPC array just comes back empty:
```
{ Vpcs: [] }
```
I've looked at the ec2 object that I create at the start and it appears ok. If you need any more info about this then please get in touch and I'll be happy to help as much as I can.

**Comment by 3y3** [comment]
Thanks for this issue @antonosmond , how you can see `node-inspector` 0.11.0 was released with new feature - **Network debugging**.
So, we wrap a part of code in app to grab information about http requests. This is possible that this new feature is not so stable as expected.
Can you please install `node-inspector@0.10.*` to confirm that problem is in Network wrapping. This is first important step.
The second step - can you please inspect what's wrong with wrapping:
1. You can look on headers and data which sended to AWS server (fiddler or wireshark will be helpful)
2. You also can look that node-inspector says about sended data

**Comment by antonosmond** [comment]
@3y3 I installed node-inspector@0.10.2 and it works correctly so I can confirm the issue is only in v0.11.*.
I'll try and find time to inspect the requests in the next day or two and let you know if I find anything. Thanks!

**Comment by dayglo** [comment]
Going back to node-inspector@0.10.2 also worked for me - dynamodb was failing with CRC errors.

**Comment by 3y3** [comment]
Can you provide some small example where I can reproduce this problem?
To avoid this problem you can also update NI to 0.12.* and start it with `--no-inject.network` flag.
This prevents NI from wrapping app `requests`.

Please he;p me to understand what is wrong with current `request` wrapper, where we miss the data.

**Comment by markacola** [comment]
@3y3 starting with --no-inject.network flag doesn't work for me.

Dropping back to node-inspector@0.10.2 does.

**Comment by zephor** [comment]
I'm getting this same problem with dynamodb and the CRC checking failures mentioned by @dayglo

**Comment by scholarvadim85** [comment]
I've tested aws-sdk with node-inspector 0.12.3 and it does not act.
For example, when I try s3.listObjects, the data goes to wrong by node --debug app.js while node app.js acts correctly. 
If I check it with 0.10.2 version, it's good for aws-sdk, but it's uncomfortable in other sides.
So I hope to know if you didn't consider aws-sdk debug in this newest version(0.12.3) or else if I can get any patch from you, node-inspector creators.

**Comment by adambiggs** [comment]
+1

I just ran into this issue as well.

**Comment by glennschler** [comment]
@AdityaManohar  I have not been able to investigate enough to give details to the aws team, but I believe this is an aws-sdk-js issue. I have not forked to run all tests with an aws-sdk change which works for me.

The nodejs.org stream document for [readable.read([size])](https://nodejs.org/api/stream.html#stream_readable_read_size) states: 'This method should _only_ be called in paused mode', yet code inspection of [aws-sdk lib/event_listeners.js line 192](https://github.com/aws/aws-sdk-js/blob/master/lib/event_listeners.js#L192) shows this `read()` call. When placing a `isPause()` check at line 190, my logs show the stream is _never_ in paused mode. So, I interpret this to mean that a ['data' event handler](https://nodejs.org/api/stream.html#stream_event_data) should be used instead.

This is my local code change which I have not fully tested for all cases. Though have been using successfully for my project  when debugging with node-inspector, and when not debugging. I have made this change to my local `node_modules/aws-sdk/lib/event_listeners.js` by adding `httpResp.isPaused()` at line 190 on debian and OSX.
```javascript
// Add check for isPaused() before listening for the 'readable' event
// If in Flowing mode (not paused) then existing code will listen for 'data' event instead
if (httpResp.isPaused() && AWS.HttpClient.streamsApiVersion === 2) {
```

**Comment by simonkontschak** [comment]
+1 same for me

**Comment by antoinegoutagny** [comment]
Upgrading to 0.12.3 seems to do the trick for me (I had empty responses from Dynamo).

**Comment by adambiggs** [comment]
I'm running 0.12.3 and still having issues.

**Comment by Yontih** [comment]
+1 same for me

**Comment by gregorskii** [comment]
+1 caused much grief on a current build. Was thinking it was intermittent, found out it was NI.

**Comment by 3y3** [comment]
Should be fixed in 0.12.5
Feel free to reopen this issue if it still exist.
